cluster/addons/storage-crds/csidriver.yaml

```diff
@@ -25,4 +25,11 @@ spec:
               description: Indicates this CSI volume driver requires additional pod
                 information (like podName, podUID, etc.) during mount operations.
               type: string
```

Comment:
> // metadata.Name indicates the name of the CSI driver that this object
> // refers to; it MUST be the same name returned by the CSI GetPluginName()

* what are the format restrictions on the value returned by the CSI `GetPluginName()` function?
* are all CSI driver names valid custom resource names?
* are all custom resource names valid CSI driver names?

Reply:
valid version is "v1", will add validation for that

GetPluginName should be GetPluginInfo
any CSI driver name is a valid custom resource name but not vice versa because:
1) CSI driver name can have length 63, object 253
2) CSI driver name can have uppercase chars, object cannot
In practice this means if the feature is enabled and the driver has uppercase we will fail here https://github.com/kubernetes/kubernetes/blob/a3ccea9d8743f2ff82e41b6c2af6dc2c41dc7b10/pkg/volume/csi/csi_plugin.go#L544 and mount/attach won't work anymore.

GetPluginInfoResponse.name:
```
  // The name MUST follow domain name notation format
  // (https://tools.ietf.org/html/rfc1035#section-2.3.1). It SHOULD
  // include the plugin's host company name and the plugin name,
  // to minimize the possibility of collisions. It MUST be 63
  // characters or less, beginning and ending with an alphanumeric
  // character ([a-z0-9A-Z]) with dashes (-), dots (.), and
  // alphanumerics between. This field is REQUIRED.
```

Reply:
nvm, mount/attach will work, we just won't find the driver object. `driver`  in ` csiDriver, err := p.csiDriverLister.Get(driver) ` comes from the CSIVolumeSource.Driver so we can say "MUST be the **lowercased** same name returned by the CSI GetPluginName()" but the `driver` var will need to be lowercased if it isn't already

Reply:
addressed in https://github.com/kubernetes/kubernetes/pull/70367/commits/f59611d72bc9ad78735f6e2efd6862dfbae562ce